README.md

```diff
@@ -1,149 +1,194 @@
+From here you can use our rules to search for issues in your codebase:
 
-```bash
-docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --help
+```
+$ cd /path/to/code
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --config https://r2c.dev/default-r2c-checks
 ```
 
-### Config Files
-
-#### Format
+You can also [create your own rules](docs/config/advanced.md):
 
-See [config/advanced.md](docs/config/advanced.md) for example configuration files and details on the syntax.
+```
+$ cd /path/to/code
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --generate-config
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep
+```
 
-#### sgrep Registry
+## Configuring
 
-[r2c](https://r2c.dev) provides a [registry](https://github.com/returntocorp/sgrep-rules) of config files tuned using our [analysis platform](https://app.r2c.dev) on thousands of repositories. To use:
+For simple patterns use the `--lang` and `--pattern` flags. This mode of
+operation is useful for quickly iterating on a pattern on a single file or
+folder:
 
-```bash
-sgrep --config r2c
+```
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --lang javascript --pattern 'eval(...)' path/to/file.js
 ```
 
-### Default
-
-Default configs are loaded from `.sgrep.yml` or multiple files matching `.sgrep/**/*.yml` and can be overridden by using `--config <file|folder|yaml_url|tarball_url|registy_name>`
+To fine-tune your searching, specify the `--help` flag:
 
-## Design
+```
+$ docker run --rm returntocorp/sgrep --help
+```
 
-Sgrep has a design philosophy that emphasizes simplicity and a single pattern being as expressive as possible:
+### Configuration Files
 
-1. **Use concrete code syntax:** easy to learn
-2. **Metavariables \($X\)**: abstract away code
-3. **'...' operator:** abstract away sequences
-4. **Knows about code equivalences:** one pattern can match many equivalent variations on the code
-5. **Less is more:** abstract away additional details
+For advanced configuration use the `--config` flag. This flag automagically
+handles a multitude of input types:
 
-## Patterns
+* `--config <file|folder|yaml_url|tarball_url|registy_name>`
 
-Patterns are snippets of code with variables and other operators that will be parsed into an AST for that language and will be used to search for that pattern in code. See [docs/config/simple.md](docs/config/simple.md) for full documentation.
+In the absense of this flag, a default configuration is loaded from `.sgrep.yml`
+or multiple files matching `.sgrep/**/*.yml`.
 
-### Metavariables
+#### Operators
 
-`$X`, `$FOO`, `$RETURNCODE` are all examples of metavariables. You can referance them later in your pattern and `sgrep` will ensure they match. **Metavariables can only contain uppercase ASCII characters**; `$x` and `$SOME_VALUE` are not valid metavariables.
+Configuration files make use of two primary operators:
+
+* **Metavariables like $X, $LIST, or $METAVARIABLE.** Metavariable names can
```

Comment:
I find it confusing to have $LIST or $METAVARIABLE as examples, they make me think there is something special about those words. Also found it helpful to have examples of the invalid metavars. Thoughts @mschwager ?

Reply:
Is it the fact that they're _any_ word, or those specific words? I think there's value in using words in general, but they don't have to be those words. E.g. word names can help you keep track of what the variable is:

```yaml
pattern-not-inside: |
    $USERS = request.get("users")
    ...
    db.remove($USERS)
```

I think this is slightly more self-documenting than something like `$X`. Does something like "**Metavariables like $X, $WIDGET, or $USERS.**" make more sense?

Definitely agree on the invalid examples - I'll add those back in.

Reply:
@mschwager yes and thanks for guessing at what I meant! I think it helps to have something like "Metavariables like $X, $WIDGET, or $USERS."